src/main/scala/beam/agentsim/agents/PersonAgent.scala

```diff
@@ -587,15 +614,10 @@ class PersonAgent(
         val nextCoord = nextActivity(data).get.getCoord
```

Comment:
Now that I look at it, I wonder if the definition of `modeOfNextLeg` above can be replaced with

`val modeOfNextLeg = _experiencedBeamPlan.getStrategy(nextAct, classOf[ModeChoiceStrategy])`

Reply:
Replayed in the comment bellow.